modifytags/gomodifytags.go

```diff
@@ -0,0 +1,830 @@
+package gomodifytags
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"io"
+	"io/ioutil"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/fatih/camelcase"
+	"github.com/fatih/structtag"
+	"golang.org/x/tools/go/buildutil"
+)
+
+// structType contains a structType node and it's name. It's a convenient
+// helper type, because *ast.StructType doesn't contain the name of the struct
+type structType struct {
+	name string
+	node *ast.StructType
+}
+
+// output is used usually by editors
+type output struct {
+	Start  int      `json:"start"`
+	End    int      `json:"end"`
+	Lines  []string `json:"lines"`
+	Errors []string `json:"errors,omitempty"`
+}
+
+// Config defines how tags should be modified
+type Config struct {
+	File     string
+	Output   string
+	Quiet    bool
+	Write    bool
+	Modified io.Reader
+
+	Offset     int
+	StructName string
+	FieldName  string
+	Line       string
+	Start, End int
+	All        bool
+
+	Fset *token.FileSet
+
+	Remove        []string
+	RemoveOptions []string
```

Comment:
Would it be more useful for these to be a `map[string][]string`, keyed by tag name?